Sprint_14

maxzim121 opened the pull request:
Добрый вечер! Я случайно закоммитил изменения в главную ветку main. Чтобы сделать pull request создал дополнительную ветку с кодом до коммитов. Надеюсь у вас получится проверить работу. Спасибо!

maxzim121 committed:
Свёрстан Launch Screen приложения.
На главном экране — две вкладки: «Трекеры» и «Статистика».
Свёрстан главный экран «Трекеры»
Реализованы экраны создания трекера.
Реализован экран задания расписания для трекера

maxzim121 committed:
Недостающие файлы

helecta commented on Tracker/CreatingTracker/TrackerCreationViewController.swift:
```diff
@@ -0,0 +1,126 @@
+
+final class TrackerCreationViewController: UIViewController, TrackerCreationDelegate {
+    
+    private var privichkaViewController = PrivichkaViewController()
```
`PrivichkaViewController` - обычно приянто называть не транслитом, а на англ языке, т.е лучше было бы назвать `HabbitViewController`

helecta commented on Tracker/CreatingTracker/TrackerCreationViewController.swift:
```diff
@@ -0,0 +1,126 @@
+    
+    private var privichkaViewController = PrivichkaViewController()
+    
+    weak var delegate: ReloadDataDelegate?
```
Стоит организовывать код следующим образом:

- Публичные переменные
- IBOutlet
- Приватные переменные
- Публичные методы(сначала `init`ы, `override`ы, и потом остальные)
- Приватные методы
- IBAction

Подробнее почитать можно тут: [Организация кода](https://github.com/tutu-ru/swift-style-guide#Организация-кода)

helecta commented on Tracker/Helpers/RecordManager.swift:
```diff
@@ -0,0 +1,34 @@
+    
+    
+    
+    static let shared = RecordManager()
```
Предполагается, что ImagesListService будет использоваться в одном экземпляре, т.е. через shared. Тогда стоит добавить приватный init. Так мы реализуем паттерн  [singleton](https://www.hackingwithswift.com/example-code/language/what-is-a-singleton) .

helecta commented on Tracker/MainScreen/SupplementaryView.swift:
```diff
@@ -0,0 +1,31 @@
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
```
Обычно в продакшене не используют fatalError, советую заменить на assertionFailure, тут подробнее:
 [https://www.hackingwithswift.com/plus/intermediate-swift/understanding-assertions](https://www.hackingwithswift.com/plus/intermediate-swift/understanding-assertions)

helecta commented on Tracker/MainScreen/TrackersCollectionViewCell.swift:
```diff
@@ -0,0 +1,160 @@
+
+final class TrackersCollectionViewCell: UICollectionViewCell {
+    
+    private var count: Int = 0
```
Стоит организовывать код следующим образом:

- Публичные переменные
- IBOutlet
- Приватные переменные
- Публичные методы(сначала `init`ы, `override`ы, и потом остальные)
- Приватные методы
- IBAction

Подробнее почитать можно тут: [Организация кода](https://github.com/tutu-ru/swift-style-guide#Организация-кода)

helecta commented on Tracker/Privichka/PrivichkaViewController.swift:
```diff
@@ -0,0 +1,214 @@
+    private let recordManager: RecordManagerProtocol = RecordManager.shared
+    private var categoryName = "Категория"
+    
+    var raspisaniePrivichki: [WeekDay] = []
```
Обрати внимание, как нужно организовывать код 

> Стоит организовывать код следующим образом:
> 
> Публичные переменные
> IBOutlet
> Приватные переменные
> Публичные методы(сначала initы, overrideы, и потом остальные)
> Приватные методы
> IBAction
> Подробнее почитать можно тут: [Организация кода](https://github.com/tutu-ru/swift-style-guide#%D0%9E%D1%80%D0%B3%D0%B0%D0%BD%D0%B8%D0%B7%D0%B0%D1%86%D0%B8%D1%8F-%D0%BA%D0%BE%D0%B4%D0%B0)

helecta commented on Tracker/Privichka/ButtonCellView.swift:
```diff
@@ -0,0 +1,59 @@
+        label.font = .systemFont(ofSize: 17)
+        label.textColor = .black
+        secondaryLabel.font = .systemFont(ofSize: 17)
+        secondaryLabel.textColor = .gray
```
Серый, который в макете не совпадает с системным серым, поэтому стоит добавить в assets цвет ypGrаy

maxzim121 committed:
Теперь трекер нельзя отметить выполненным для будущей даты. Счётчик выполненных дней обновляется при переходе на другой день. При нажатии на кнопку "Создать" создаётся новый трекер.
P.S.: Начал работу над 15 спринтом, добавились файлы не влияющие на работу программы согласно 14 спринту.

maxzim121 commented:
Лилия, добрый день! Спасибо за ревью!
Комментарии касающиеся экрана создания трекера исправлю при добавлении CollectioinView цветов и смайликов в первом задании 15 спринта.

maxzim121 committed:
Убрал лишние print не влияющие на работу программы
